### PR TITLE
Feat dli queue

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
relate to #1256  : add DLI queues to the go SDK ;

The list service returns more information, so switch the get interface to list

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/dli/v1/queues/results.go [hunk 1]
```diff
@@ -68,7 +76,53 @@ type Queue struct {
...
-	Tags []tags.ResourceTag `json:"tags"`
+	/**
+	The resource type to which the queue belongs.
+	VM: ECF cluster
+	Container: containerized cluster (k8s)
+	**/
+	QueueResourceType string `json:"queue_resource_type"`
+}
+
+type Queue4Get struct {
```
Comment (ShiChangkuo) [inline openstack/dli/v1/queues/results.go:87]: defined but not used.

File: openstack/dli/v1/queues/requests.go [hunk 2]
```diff
@@ -90,9 +91,27 @@ func Delete(c *golangsdk.ServiceClient, queueName string) (r DeleteResult) {
 }
 
 func Get(c *golangsdk.ServiceClient, queueName string) (r GetResult) {
-	result := new(Queue)
+	listResult := new(ListResult)
+
 	reqOpt := &golangsdk.RequestOpts{OkCodes: []int{200}}
-	_, r.Err = c.Get(resourceURL(c, queueName), &result, reqOpt)
-	r.Body = result
+	_, r.Err = c.Get(queryAllURL(c), &listResult, reqOpt)
```
Comment (ShiChangkuo) [inline openstack/dli/v1/queues/requests.go:97]: @hmbig2 why we change the API URI?

**Review by ShiChangkuo — CHANGES_REQUESTED** [review]

File: openstack/dli/v1/queues/requests.go (hunk 2 above)
Comment (hmbig2) [reply openstack/dli/v1/queues/requests.go:127]: The list URI returns more information

File: openstack/dli/v1/queues/results.go (hunk 1 above)
Comment (hmbig2) [reply openstack/dli/v1/queues/results.go:87]: add the new GET function

**Pull request merged**